aws/ec2metadata/api.go

```diff
@@ -20,6 +24,39 @@ func (c *EC2Metadata) GetMetadata(p string) (string, error) {
 	return output.Content, req.Send()
 }
 
+// GetDynamicData uses the path provided to request
+func (c *EC2Metadata) GetDynamicData(p string) (string, error) {
+	op := &request.Operation{
+		Name:       "GetMetadata",
```

Comment:
`Name`'s value should be `GetDynamicData` in this case. Its not directly used in the HTTP request, but is useful for logging.